decisions/20240313-try-w3up.md

```diff
@@ -0,0 +1,169 @@
+
+### Does this project have any dependencies?
+
+#### dependency on collaboration with nft.storage
+
+This project can't succeed without a sponsor from nft.storage and willingness to try out the proposed changes to the nft.storage codebase in a production-like environment. If we have this, we can help prove out nft.storage storing data via web3.storage and relying on web3.storage productized services for serving on ipfs and filecoin dealmaking.
+
+## Alternative Solutions
+
+I don't know of any alternate solutions I can link to that provide a path to servicing nft.storage's web3 storage needs without relying on dagcargo. If you know of some, please add them here.
+
+### Use w3-filecoin Directly from nft.storage (without w3up)
+
+The subsystem of w3up that does filecoin dealmaking is called [w3-filecoin](https://github.com/web3-storage/specs/blob/main/w3-filecoin.md) and has an API that nft.storage could call directly (just like other parts of w3up call it directly).
+
+@vasco-santos proposed this in [api.nft.storage onboarding to w3filecoin](https://hackmd.io/0K9xlcQRTp6pLps1bffxHQ).
+
+@vasco-santos has weighed in on the tradeoffs of that proposal vs the decision documented here (emphasis added):
+
+> w3filecoin was designed for multiple storefronts, so w3up/others, and my idea was to not go through w3up to avoid mixing the data at that level. And this together with the separate aggregation process for both. The general sense though is that **using w3up makes more sense on a perspective on having nft.storage as a user of w3up in the long run**. But if the expectation we have is it to be 3 months, making it a client of w3filecoin, while continuing to write to old bucket and so on, would make more sense to me.
+>
+> But after talking with Alan, Hannah, Reid, we opted on the w3up lane, and that is why I say that it seems out of the table. **I generally agree with using w3up because it is simpler if we won't care about splitting aggregation + have nft.storage in the long run as a client of w3up**
+
+If we determine it is not desirable for nft.storage to use the entirety of w3up's normal functionality, we may still pursue this alternative solution.
+
+## Implementation and Rollout Plan
+
+Prototype Happy path
+
+- [x] @gobengo set up nft.storage localdev to be able to hack on packages/api
+- [ ] nft.storage/api: can be configured with `W3_NFTSTORAGE_PRINCIPAL` and `W3_NFTSTORAGE_PROOF` and use them to get a w3up-client at runtime
+- [ ] nft.storage/api: modify POST /upload/ handler to store using w3up-client@12
+- [ ] nft.storage/api: add feature switch requiring allowlist for new uploads to store in web3.storage
+
+Rollout
+
+- [ ] release modified nft.storage to staging (production ok too) but with web3.storage disabled-by-default via feature switch
```

Comment:
Elizabeth is planning to release a new Website/console which is currently in development by some contractors. We need to understand what she is changing and what the impact on this roll out plan will be.

Reply:
this is an API change, so as soon a new website/console uses same API everything would be fine. But from what I know (or in other words, pieces I put together from all the meetings I attended), the new website/console they are putting together would be a complete new thing that would mean all of this is deprecated

Reply:
> We need to understand what she is changing and what the impact on this roll out plan will be.

sounds like a good thing for next sprint